src/routeGeneration/routeGenerator.ts

```diff
@@ -31,7 +31,7 @@ export class RouteGenerator {
   constructor(private readonly metadata: Tsoa.Metadata, private readonly options: RoutesConfig, private readonly minimalSwaggerConfig: SwaggerConfigRelatedToRoutes) { }
 
   public async GenerateRoutes(middlewareTemplate: string, pathTransformer: (path: string) => string) {
-    const fileName = `${this.options.routesDir}/routes.ts`;
```

Comment:
Please add validation here that `this.options.routesDir` is actually a directory. I think the only way that we can do that is check:
* that it doesn't end in `.ts` or `.js` or .json` etc.
  * If it does end in an extension, we should throw an error with the message of `routesDir must be a directory but it was given the path of a file. Perhaps you were looking to use the routesFileName config option?`

Reply:
Makes sense.
Wouldn't it make more sense to just [directly check](https://stackoverflow.com/a/15630832/10434371) if a path is a directory?